### PR TITLE
[GO][Client] return GenericOpenAPIError instead of error in Execute()

### LINKED ISSUE #8136: [GO][BUG] *_api.go CRUD function return error instead of GenericOpenAPIError
#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

In go client, *_api.go files, `Execute()` return 3 parameters, the last one being error. Since this of type `error`, response body is not received. Instead, only the Status code is sent. Plus once the response is read in the `Execute()`, it cannot be read again. (Well, this is how it works) Sending the second parameter `nethttp.Response` is of not much use because it has the body missing.

In all, the response from server in case of a failure is never sent back to the user.

##### openapi-generator version

v5.0.0-beta3 

##### Suggest a fix

Use of `GenericOpenAPIError` as return type of 3rd parameter in the `Execute` function. Use `NopCloser` from `ioutil` to repopulate the body of response, so that it can be used later.

### PR DESCRIPTION
**Pull request by code-lucidal58**
fix #8136 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@antihax (2017/11) @grokify (2018/07) @kemokemo (2018/09) @bkabrda (2019/07)

### CODE REVIEW

**Comment by code-lucidal58** [comment]
Code snippet used to test the generated client code
```go
result, _, err := r.Execute()
if err.Error() != "" {
	return fmt.Errorf("error occurred while updating object: %s Response: %s", err.Error(), string(err.Body()))
}
```
Output: 
```
Error: error occurred while updating VnicEthIf object: 400 Bad Request Response: <response body as it is >
```

Earlier Output:
```
Error: error occurred while updating VnicEthIf object: 400 Bad Request Response: 
```
When printing the error `err` only because `error` does not support `.Body()`, `GenerateOpenAPIError` does.

File: modules/openapi-generator/src/main/resources/go/api.mustache [hunk 1]
```diff
@@ -3,6 +3,7 @@ package {{packageName}}
 
 {{#operations}}
 import (
+  "bytes"
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/go/api.mustache:6]: @code-lucidal58 what about using a tab instead?

Comment (code-lucidal58) [reply modules/openapi-generator/src/main/resources/go/api.mustache:6]: I use tabs. And looks fine in Atom and Intellij. Not sure why it looks different here. Let me try again

Comment (code-lucidal58) [reply modules/openapi-generator/src/main/resources/go/api.mustache]: done

**Comment by wing328** [comment]
Looks like there are syntax errors causing build failure:

```
# _/home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/go/go-petstore
go-petstore/api_another_fake.go:95:2: syntax error: non-declaration statement outside function body
FAIL	_/home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/go [build failed]
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2.1:exec (go-test) on project GoPetstore: Command execution failed. Process exited with an error: 2 (Exit value: 2) -> [Help 1]
```

**Comment by code-lucidal58** [comment]
All errors fixed @wing328

**Comment by code-lucidal58** [comment]
Not able to figure out why the circleci build fails.

**Comment by wing328** [comment]
I think you will need to update the test as well. Here is the error reported by the CI:
```
Download https://repo1.maven.org/maven2/org/junit/platform/junit-platform-engine/1.4.0/junit-platform-engine-1.4.0.jar
Download https://repo1.maven.org/maven2/org/junit/jupiter/junit-jupiter-api/5.4.0/junit-jupiter-api-5.4.0.jar
Download https://repo1.maven.org/maven2/org/junit/platform/junit-platform-commons/1.4.0/junit-platform-commons-1.4.0.jar
:processResources UP-TO-DATE
:classes
:compileTestJava UP-TO-DATE
:compileTestGroovy
:processTestResources UP-TO-DATE
:testClasses
:test

BUILD SUCCESSFUL

Total time: 22.356 secs
# _/home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/go/mock
mock/mock_api_pet.go:20:29: cannot use m (type *MockPetApi) as type petstore.PetApi in field value:
	*MockPetApi does not implement petstore.PetApi (wrong type for AddPetExecute method)
		have AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, error)
		want AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, petstore.GenericOpenAPIError)
mock/mock_api_pet.go:28:32: cannot use m (type *MockPetApi) as type petstore.PetApi in field value:
	*MockPetApi does not implement petstore.PetApi (wrong type for AddPetExecute method)
		have AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, error)
		want AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, petstore.GenericOpenAPIError)
mock/mock_api_pet.go:36:39: cannot use m (type *MockPetApi) as type petstore.PetApi in field value:
	*MockPetApi does not implement petstore.PetApi (wrong type for AddPetExecute method)
		have AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, error)
		want AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, petstore.GenericOpenAPIError)
mock/mock_api_pet.go:44:37: cannot use m (type *MockPetApi) as type petstore.PetApi in field value:
	*MockPetApi does not implement petstore.PetApi (wrong type for AddPetExecute method)
		have AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, error)
		want AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, petstore.GenericOpenAPIError)
mock/mock_api_pet.go:52:33: cannot use m (type *MockPetApi) as type petstore.PetApi in field value:
	*MockPetApi does not implement petstore.PetApi (wrong type for AddPetExecute method)
		have AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, error)
		want AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, petstore.GenericOpenAPIError)
mock/mock_api_pet.go:60:32: cannot use m (type *MockPetApi) as type petstore.PetApi in field value:
	*MockPetApi does not implement petstore.PetApi (wrong type for AddPetExecute method)
		have AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, error)
		want AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, petstore.GenericOpenAPIError)
mock/mock_api_pet.go:68:40: cannot use m (type *MockPetApi) as type petstore.PetApi in field value:
	*MockPetApi does not implement petstore.PetApi (wrong type for AddPetExecute method)
		have AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, error)
		want AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, petstore.GenericOpenAPIError)
mock/mock_api_pet.go:76:33: cannot use m (type *MockPetApi) as type petstore.PetApi in field value:
	*MockPetApi does not implement petstore.PetApi (wrong type for AddPetExecute method)
		have AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, error)
		want AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, petstore.GenericOpenAPIError)
mock/mock_api_pet.go:84:49: cannot use m (type *MockPetApi) as type petstore.PetApi in field value:
	*MockPetApi does not implement petstore.PetApi (wrong type for AddPetExecute method)
		have AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, error)
		want AddPetExecute(petstore.ApiAddPetRequest) (*http.Response, petstore.GenericOpenAPIError)
FAIL	_/home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/go [build failed]
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2.1:exec (go-test) on project GoPetstore: Command execution failed. Process exited with an error: 2 (Exit value: 2) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :GoPetstore
2020-12-13 15:52:05.344:INFO::main: Logging initialized @106ms
```

**Comment by wing328** [comment]
All tests passed. If no further feedback/question from anyone, I'll merge it on coming Wed.
